pegen/parse_string.c

```diff
@@ -578,9 +578,14 @@ fstring_compile_expr(Parser *p, const char *expr_start, const char *expr_end,
 
     if (asdl_seq_LEN(mod->v.Module.body) == 0) {
         raise_syntax_error(p, "f-string: empty expression not allowed");
+        goto exit;
     }
 
     stmt_ty expr = asdl_seq_GET(mod->v.Module.body, 0);
+    if (asdl_seq_LEN(mod->v.Module.body) != 1 || expr->kind != Expr_kind) {
```

Comment:
This cannot be here as there is a check later for `mod == NULL`. If `mod` is `NULL` then `asdl_seq_LEN(mod->v.Module.body)` will segfault.

Reply:
That's right, but won't `stmt_ty expr = asdl_seq_GET(mod->v.Module.body, 0);` segfault as well, then?

Reply:
🤦‍♂ Yep, that was wrong as well. We need to do the `NULL` check before we do anything else.

Reply:
Oh, there is a null check for `mod` on line 575. This only gets reached, in case it isn't NULL.

Reply:
🤦‍♂ 🤦‍♂ Note to self: do not review code in public transport 😉